src/patients/appointments/AppointmentsList.tsx

```diff
@@ -1,37 +1,46 @@
 import { Button, Table, Spinner, Alert } from '@hospitalrun/components'
 import format from 'date-fns/format'
-import React, { useEffect } from 'react'
-import { useSelector, useDispatch } from 'react-redux'
+import React from 'react'
 import { useHistory } from 'react-router-dom'
 
 import useAddBreadcrumbs from '../../page-header/breadcrumbs/useAddBreadcrumbs'
-import { fetchPatientAppointments } from '../../scheduling/appointments/appointments-slice'
 import useTranslator from '../../shared/hooks/useTranslator'
-import { RootState } from '../../shared/store'
+import usePatientsAppointments from '../hooks/usePatientAppointments'
 
 interface Props {
   patientId: string
 }
 
 const AppointmentsList = (props: Props) => {
-  const dispatch = useDispatch()
   const history = useHistory()
   const { t } = useTranslator()
 
   const { patientId } = props
-  const { appointments } = useSelector((state: RootState) => state.appointments)
+
+  const { data, status } = usePatientsAppointments(patientId)
 
   const breadcrumbs = [
     {
       i18nKey: 'scheduling.appointments.label',
       location: `/patients/${patientId}/appointments`,
     },
   ]
+
   useAddBreadcrumbs(breadcrumbs)
 
-  useEffect(() => {
-    dispatch(fetchPatientAppointments(patientId))
-  }, [dispatch, patientId])
+  if (data === undefined || status === 'loading') {
+    return <Spinner color="blue" loading size={[10, 25]} type="ScaleLoader" />
```

Comment:
We've got a `Loading` component in this project that keeps loading spinners consistent for this case.

Reply:
I updated this my latest commit. please let me know what think of it